### DISCUSSION TITLE
Could not create a Policy

### DISCUSSION THREAD

**Discussion by Akshay-a-j** [eclipse-edc/Samples#88] [Q&A] [answered]
Hello,

I attempted to replicate the [transfer-06-consumer-pull-http](https://github.com/eclipse-edc/Samples/tree/main/transfer/transfer-06-consumer-pull-http) scenario but encountered difficulties in creating a policy. 

I had previously succeeded in doing so a few days ago. However, after cloning the Samples repository again and attempting to replicate the process, I encountered an error when trying to create the policy using the provided set of commands.

curl -d '{
    "@context": {
      "edc": "https://w3id.org/edc/v0.0.1/ns/",
      "odrl": "http://www.w3.org/ns/odrl/2/"
    },
    "@id": "cPolicy",
    "policy": {
      "@type": "set",
      "odrl:permission": [
        {
          "odrl:action": "USE",
          "odrl:constraint": [
            {
              "odrl:leftOperand": "ids:absoluteSpatialPosition",
              "odrl:operator": "EQ",
              "odrl:rightOperand":  "eu"
            }]
        }
      ],
      "odrl:prohibition": [],
      "odrl:obligation": []
    }
  }' -H 'content-type: application/json' http://localhost:19193/management/v2/policydefinitions \
  -s | jq

and the error message was: 

[
  {
    "message": "http://www.w3.org/ns/odrl/2/Operator property '@id' was missing",
    "type": "InvalidRequest",
    "path": null,
    "invalidValue": null
  },
  {
    "message": "ARRAY must be EQ, NEQ, GT, GEQ, LT, LEQ, IN, HAS_PART, IS_A, IS_ALL_OF, IS_ANY_OF, or IS_NONE_OF but was: unknown",
    "type": "InvalidRequest",
    "path": null,
    "invalidValue": null
  }
]

Could someone kindly inform me of the potential cause of this issue?

Thank you

**Comment by Akshay-a-j** [comment]
Update: We can create policy if "permission":[ ] is empty. However, it fails to create a policy when above given permission is included in the command.

**Comment by ndr-brt** [comment, marked as answer]
please try to add the odrl.jsonld file as context:
```
{
  "@context": {
    "edc": "https://w3id.org/edc/v0.0.1/ns/",
    "odrl": "http://www.w3.org/ns/odrl/2/"
  },
  "@id": "cPolicy",
  "policy": {
    "@context": "http://www.w3.org/ns/odrl.jsonld",
    "@type": "set",
    "permission": [
      {
        "action": "USE",
        "constraint": [
          {
            "leftOperand": "ids:absoluteSpatialPosition",
            "operator": "EQ",
            "rightOperand": "eu"
          }]
      }
    ],
    "prohibition": [],
    "obligation": []
  }
}
```

(PS: please use triple backtick block to monospaced code, otherwise it will become unreadable ;) )
